src/builder.cpp

```diff
@@ -345,54 +345,17 @@ void Builder::set_entity_brush_common(int idx, Node3D* node, LMEntity& ent)
 
 	} else if (node->is_class("CollisionObject3D")) {
 		// If it's not a dynamic body, we can just use a concave trimesh collider
-		need_collider = ColliderType::Static;
+		need_collider = ColliderType::Mesh;
 		need_collider_shape = ColliderShape::Concave;
 	}
 
-	if (node->is_class("PhysicsBody3D")) {
-		need_visual = true;
-	}
-
-	//TODO: Modify visual or collider from ent properties
-
 	// Stop if we don't need to do anything
 	if (!need_visual && need_collider == ColliderType::None) {
 		UtilityFunctions::printerr("Brush entity class has no need for visual nor collision: ", node->get_class());
 		return;
 	}
 
-	// If we need at least visual, create mesh instance for each texture, and create colliders in the process
-	if (need_visual) {
-		for (int i = 0; i < m_map->texture_count; i++) {
-			auto& tex = m_map->textures[i];
-			build_texture_mesh(idx, tex.name, ent, node, need_collider, need_collider_shape);
-		}
-		return;
-	}
-
-	// We don't need visual, we only need collision
-	if (need_collider != ColliderType::None) {
-		// Gather brush surfaces
-		LMSurfaceGatherer surf_gather(m_map);
-		surf_gather.surface_gatherer_set_entity_index_filter(idx);
-		surf_gather.surface_gatherer_run();
-
-		auto& surfs = surf_gather.out_surfaces;
-		if (surfs.surface_count == 0) {
-			return;
-		}
-
-		for (int i = 0; i < surfs.surface_count; i++) {
-			auto& surf = surfs.surfaces[i];
-			if (surf.vertex_count == 0) {
-				continue;
-			}
-
-			// Create the mesh
-			auto mesh = create_mesh_from_surface(surf);
-			add_collider_from_mesh(node, mesh, need_collider_shape);
-		}
-	}
+	MeshInstance3D* mesh_instance = build_entity_mesh(idx, ent, node, ColliderType::Mesh, need_collider_shape);
```

Comment:
I think you want to use `need_collider` here?

Reply:
True, I'm dumb